tempodb/compactor.go

```diff
@@ -66,6 +66,11 @@ const (
 
 // todo: pass a context/chan in to cancel this cleanly
 func (rw *readerWriter) compactionLoop() {
+	compactionCycle := DefaultCompactionCycle
```

Comment:
We may set the default value in `RegisterFlagsAndApplyDefaults`, and then we can use the value here.

Reply:
I'm not sure I follow what you mean, but isn't the default  already being set in `RegisterFlagsAndApplyDefaults` ?

https://github.com/willdot/tempo/blob/6b83318b934f4c961fc5e962ac448e3140202254/modules/compactor/config.go#L30

Reply:
Oh you're right, I read to fast.

Reply:
Could we reference the `rw.compactorCfg.CompactionCycle` below in place of the local?  Does a `0` compaction cycle mean something, or I suppose in that case a config value of `0` would be actually `30`, which could be confusing.

Reply:
If 0 is passed into a new ticker, you get a panic `non-positive interval for NewTicker`

Which is why I use the default if the config is set to 0. 

Thinking about it, this either means it's not been configured (valid) or it's been configured as 0 (also valid). Is there somewhere we can document to users that it can't be 0 and that the default of 30 seconds will be used if it is?

To make the code clearer I could either drop a comment, or extract it out into a helper function, something like 

`getCompactionCycleFromConfigOrDefault(rw.compactorCfg.CompactionCycle, DefaultCompactionCycle)`

Reply:
I'd vote for using the default of 30 seconds even if the user manually configures a value of 0 (because that doesn't make sense). We can make a note of it in our configuration docs, which are over here:
https://github.com/grafana/tempo/blob/main/docs/tempo/website/configuration/_index.md#compactor

Reply:
Added something to the configuration docs.